src/Framework/TestCase.php

```diff
@@ -105,26 +112,33 @@ public function runTest($method, array $args = NULL)
 			$data[] = $args;
 		}
 
-		$me = $this;
-		$errorHandler = function ($severity, $message) use ($me, & $prev) {
-			if (($severity & error_reporting()) === $severity) {
-				restore_error_handler();
-				$rm = new \ReflectionMethod($me, 'tearDown');
-				$rm->setAccessible(TRUE);
 
-				set_error_handler(function () {}); // mute all errors
-				$rm->invoke($me);
-				restore_error_handler();
-			}
+		if ($this->prevErrorHandler === FALSE) {
+			$me = $this;
+			$handleErrors = & $this->handleErrors;
+			$prev = & $this->prevErrorHandler;
+
+			$prev = set_error_handler(function ($severity) use ($me, & $prev, & $handleErrors) {
+				if ($handleErrors === TRUE && ($severity & error_reporting()) === $severity) {
+					$handleErrors = FALSE;
+					$rm = new \ReflectionMethod($me, 'tearDown');
+					$rm->setAccessible(TRUE);
+
+					set_error_handler(function() {});  // mute all errors
+					$rm->invoke($me);
+					restore_error_handler();
```

Comment:
This is the similar situation. User can setup own error handler in `tearDown`. But the `$handleErrors` prevents recursion now.